File: day1/part2.c
#include <stdio.h>

int main(void)
{
    int arr1[1000];
    int arr2[1000];

    int total = 0;

    FILE *file = fopen("input1.txt", "r");

    int n = 0;
    for (int i = 0; i < sizeof(arr1) / sizeof(arr1[0]); i++)
    {
        fscanf(file, "%d %d", &arr1[i], &arr2[i]);
        n++;
    }

    for (int i = 0; i < n; i++)
    {
        int count = 0;
        for (int j = 0; j < n; j++)
        {
            if (arr1[i] == arr2[j])
            {
                count += 1;
            }
        }
        total += count * arr1[i];
    }
    printf("Total: %d ", total);

    return 0;
}

File: day1/part2.py
list1 = []
list2 = []
seen_values = dict()

with open('input1.txt', 'r') as file:
    for line in file:
        id1, id2 = line.split()
        list1.append(id1)
        list2.append(id2)

for elem in list2:
    if elem in seen_values:
        seen_values[elem] += 1
    else:
        seen_values[elem] = 1

total = 0
for elem in list1:
    if elem in seen_values:
        total += seen_values[elem] * int(elem)

print(total)



        

        